testing/test_impl.hpp

```diff
@@ -186,8 +187,8 @@ void unit_test::set_buffer_ptr_array (
 
     // fill each index with its number
     T i = 0;
-    T one = 1;
-    for (size_t idx = 0; idx < size; idx++, i+one) {
+    // T one = 1;
+    for (size_t idx = 0; idx < size; idx++ /*, i+one*/) {
```

Comment:
As you show, there's no real need for increasing `i` with the loop when checking indexing. You can feel free to remove all comments if you feel it's unnecessary.